lib/elixir/src/elixir_errors.erl

```diff
@@ -98,12 +98,24 @@ snippet(InputString, Location, StartLine, StartColumn) ->
 -spec parse_error(elixir:keyword(), binary() | {binary(), binary()},
                   binary(), binary(), {unicode:charlist(), integer(), integer()}) -> no_return().
 parse_error(Location, File, Error, <<>>, {InputString, StartLine, StartColumn}) when is_list(InputString) ->
-  Message = case Error of
-    <<"syntax error before: ">> -> <<"syntax error: expression is incomplete">>;
-    _ -> <<Error/binary>>
-  end,
-  Snippet = snippet(InputString, Location, StartLine, StartColumn),
-  raise(Location, File, 'Elixir.TokenMissingError', Message, Snippet);
+  case Error of
+    <<"syntax error before: ">> ->
+      Message = <<"syntax error: expression is incomplete">>,
+      Snippet = snippet(InputString, Location, StartLine, StartColumn),
+      raise(Location, File, 'Elixir.TokenMissingError', Message, Snippet);
+
+    <<"missing terminator: ", _/binary>> ->
+      case snippet(InputString, Location, StartLine, StartColumn) of
+        #{content := <<"">>} ->
+          raise(Location, File, 'Elixir.TokenMissingError', Error);
```

Comment:
Do we want to have this check inside "missing terminator: "? Or do we want to not show the snippet every time `content := <<"">>`? Maybe we should have `snippet/4` return `nil` if the content is `<<"">>` and then in this function we simply don't care about handling the snippet?

Reply:
I think in general snippet on empty lines are not helpful, as there is nothing to point to. But also unlikely. So I think we can generalize this, yeah. Possible also rule it out if it only has whitespaces. So if `string:trim(String, leading)` is an empty binary, we skip it. WDYT?

Reply:
Agreed! Pushed a commit that follows this rule. Also added tests for this behaviour.